client/genericclient/client.go

```diff
@@ -31,12 +31,15 @@ var _ Client = &genericServiceClient{}
 
 // NewClient create a generic client
 func NewClient(destService string, g generic.Generic, opts ...client.Option) (Client, error) {
-	svcInfo := generic.ServiceInfoWithCodec(g)
+	svcInfo := generic.ServiceInfoWithGeneric(g)
 	return NewClientWithServiceInfo(destService, g, svcInfo, opts...)
 }
 
 // NewClientWithServiceInfo create a generic client with serviceInfo
 func NewClientWithServiceInfo(destService string, g generic.Generic, svcInfo *serviceinfo.ServiceInfo, opts ...client.Option) (Client, error) {
```

Comment:
What about `NewStreamingClientWithServiceInfo`?

Reply:
@jayantxie Streaming client uses the different service info, so it didn't use the old API, which means it doesn't need to add this logic.